app/controllers/application_controller/filter.rb

```diff
@@ -947,7 +947,11 @@ def adv_search_build(model)
     # Restore @edit hash if it's saved in @settings
     @expkey = :expression                                               # Reset to use default expression key
     if session[:adv_search] && session[:adv_search][model.to_s]
-      @edit = copy_hash(session[:edit])
```

Comment:
@hstastna I think you would be a good candidate to review this since you added the above line (`@edit = copy_hash(session[:edit])`)